Source/BestProjectEver/Private/Components/BPE_HealthComponent.cpp

```diff
@@ -0,0 +1,81 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "Components/BPE_HealthComponent.h"
+
+#include "Character/BPE_Enemy.h"
+#include "Net/UnrealNetwork.h"
+
+
+UBPE_HealthComponent::UBPE_HealthComponent()
+{
+	PrimaryComponentTick.bCanEverTick = false;
+
+	MaxHealth = 100.0f;
+	bIsDead = false;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void UBPE_HealthComponent::BeginPlay()
+{
+	Super::BeginPlay();
+
+	CurrentHealth = MaxHealth;
+	AActor* ActorOwner = GetOwner();
+	if (IsValid(ActorOwner) && ActorOwner->HasAuthority())
+	{
+		ActorOwner->OnTakeAnyDamage.AddDynamic(this, &UBPE_HealthComponent::HandleTakeAnyDamage);
+	}
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void UBPE_HealthComponent::HandleTakeAnyDamage(AActor* DamagedActor, float Damage, const UDamageType* DamageType,
+	AController* InstigatedBy, AActor* DamageCauser)
+{
+	if (Damage <= 0.0f || bIsDead)
+	{
+		return;
+	}
+
+	CurrentHealth = FMath::Clamp(CurrentHealth - Damage, 0.0f, MaxHealth);
+	OnHealthChange();
+	
+	if (CurrentHealth == 0.0f) 
```

Comment:
It's good to use `FMath::IsNearlyZero(CurrentHealth)` here, to avoid precision errors that make this condition false if CurrentHealth is something like 0.000000000001

Reply:
![image](https://user-images.githubusercontent.com/85327912/173107904-deff9f47-2c71-4a9e-8219-60df28e0101c.png)